Firebase.h

```diff
@@ -122,7 +131,7 @@ class FirebasePush : public FirebaseCall {
  public:
   FirebasePush() {}
   FirebasePush(const String& host, const String& auth,
-               const String& path, const String& value, HTTPClient* http = NULL);
+               const String& path, const String& value, FirebaseHttpClient* http = NULL);
```

Comment:
If wonder if instead we could invert the dependency:
i.e: make the FirebaseHttpClientEsp8266 object accept a FirebasePush/Get/etc object.

That way we could move all the network logic to the concrete type and let the user init the one that correspond to their platform to send firebase object to the wire, but the core of the library for constructing payload and parsing response would be cross platform.